### PR TITLE
Reactively wait for required conditions in tests

### PR DESCRIPTION
**Pull request by andrei-21**
It should fix sporadic CI failures of p2p tests.

### CODE REVIEW

**Review by danielgranhao — APPROVED** [review]
Cool!

File: eel/tests/p2p_connection_test.rs [hunk 1]
```diff
@@ -49,8 +46,7 @@ mod p2p_connection_test {
             setup::nigiri::wait_for_healthy_lspd();
             // TODO: Once reconnect period exposed as a config, config it with a
             //       smaller value to speedup the test.
```
Comment (gcomte) [inline eel/tests/p2p_connection_test.rs:48]: Maybe that comment is not really relevant anymore

**Comment by gcomte** [comment]
Please change the target branch from `develop` to `main`

**Comment by gcomte** [comment]
A [new PR](https://github.com/getlipa/lipa-lightning-lib/pull/319) has been created which obsoletes this one.

**Pull request closed without merging**